example/instance/vagrant/main.go

```diff
@@ -1,27 +1,29 @@
 package main
 
 import (
+	"os"
+
 	log "github.com/Sirupsen/logrus"
 	"github.com/docker/infrakit/cli"
 	"github.com/docker/infrakit/plugin/instance/vagrant"
 	instance_plugin "github.com/docker/infrakit/spi/http/instance"
 	"github.com/spf13/cobra"
-	"os"
 )
 
 func main() {
 
 	var name string
 	var logLevel int
 	var dir string
+	var template string
 
 	cmd := &cobra.Command{
 		Use:   os.Args[0],
 		Short: "Vagrant instance plugin",
 		Run: func(c *cobra.Command, args []string) {
 
 			cli.SetLogLevel(logLevel)
-			cli.RunPlugin(name, instance_plugin.PluginServer(vagrant.NewVagrantPlugin(dir)))
+			cli.RunPlugin(name, instance_plugin.PluginServer(vagrant.NewVagrantPlugin(dir, template)))
```

Comment:
Let's go ahead and read the template file here, and inject the template contents (or the `const` if the template arg was not specified).  This way we fail fast if the file doesn't exist, and avoid the branch in `instance.go`.

The next step is to accept an abstract JSON schema for the instance plugin properties (so that when i write a template i can define my own variables).  But that can be an exercise for another day.